h2o-core/src/main/java/water/network/SecurityUtils.java

```diff
@@ -96,15 +96,24 @@ public static String generateSSLConfig(SSLCredentials credentials) throws IOExce
         return generateSSLConfig(credentials, temp.getAbsolutePath());
     }
 
-    public static String generateSSLConfig(SSLCredentials credentials, String file) throws IOException {
+    static String generateSSLConfig(SSLCredentials credentials, String file) throws IOException {
         Properties sslConfig = new Properties();
         sslConfig.put("h2o_ssl_protocol", defaultTLSVersion());
-        sslConfig.put("h2o_ssl_jks_internal", credentials.jks.getLocation());
+        sslConfig.put("h2o_ssl_jks_internal", credentials.jks.name);
         sslConfig.put("h2o_ssl_jks_password", credentials.jks.pass);
-        sslConfig.put("h2o_ssl_jts", credentials.jts.getLocation());
+        sslConfig.put("h2o_ssl_jts", credentials.jts.name);
         sslConfig.put("h2o_ssl_jts_password", credentials.jts.pass);
         FileOutputStream output = new FileOutputStream(file);
-        sslConfig.store(output, "");
+        try {
+            sslConfig.store(output, "");
+            output.close();
+        } finally {
+            try {
+                output.close();
```

Comment:
doing output.close(); twice, once in finally is enough?

Reply:
It is not, this is one of the recommended approaches closing a resource if you don't use try-with-resources and don't want to lose the Exception accidentally. 

The point of the close inside of the try block is to make sure that if a close() produces an exception it is actually thrown and not suppressed. If close() inside of the finally-block generates an Exception we know it is okay to ignore because that can only happen if there was another root cause.